### DISCUSSION TITLE
Incorrect timestamp parsing

### DISCUSSION THREAD

**Discussion by paymog** [vectordotdev/vector#20959] [Q&A] [answered]
I'm having trouble parsing a timestamp correctly in vector

```
$ parse_timestamp!("2024-07-29T14:13:44.573709634Z", "%Y-%m-%dT%H:%M:%S.%fZ")
t'2024-07-29T19:13:44.573709634Z'
```

As far as I can tell, this should resolve to an hour of `14` not `19` (I'm 5 hours behind UTC) and when I test it [here](https://jeffkayser.com/projects/date-format-string-composer/index.html) I get the right output (adjusted for my local time zone)

<img width="1256" alt="image" src="https://github.com/user-attachments/assets/bf31eeaf-1d33-4357-9292-7777dff61c6a">

How do I tell vector that this string is in fact time UTC time and not my local time zone?

**Comment by jszwedko** [comment, marked as answer]
Hi @paymog ,

In your pattern `Z` is matched as a literal so the timestamp is interpreted as a timestamp without a timezone, which is why it uses the local timezone. I think you want: `parse_timestamp!("2024-07-29T14:13:44.573709634Z", "%+")` which will use the timezone from the timestamp.

**Reply by paymog** [reply]
Well I'm an idiot...

Thanks for the help!
